src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -384,18 +387,23 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
                             let zAxisMultiplier = this._scene.useRightHandedSystem? 1 : -1;
                             const indexTipPos = indexTipPose.transform.position;
                             const indexTipOrientation = indexTipPose.transform.orientation;
+                            controllerData.indexTipQuaternion.set(indexTipOrientation.x, indexTipOrientation.y, (indexTipOrientation.z * zAxisMultiplier), (indexTipOrientation.w * zAxisMultiplier));
+
                             // set positions for near pick and hover
                             if (controllerData.pickIndexMeshTip) {
                                 controllerData.pickIndexMeshTip.position.set(indexTipPos.x, indexTipPos.y, (indexTipPos.z * zAxisMultiplier));
                             }
                             if (controllerData.hoverIndexMeshTip) {
-                                controllerData.hoverIndexMeshTip.position.set(indexTipPos.x, indexTipPos.y, (indexTipPos.z * zAxisMultiplier));
+                                controllerData.hoverIndexMeshTip.position.set(indexTipPos.x, indexTipPos.y, indexTipPos.z);
                             }
+
                             // set near interaction grab ray parameters
                             let nearGrabRayLength = 0.5;
-                            let indexTipZaxisMultiplier = -1;
+                            let webxrZaxisMultiplier = -1;
+                            let indexTipOrientationVector = Vector3.Zero();

```

Comment:
Probably good to cache this too, instead of creating a new one for each controller every frame.

Reply:
this is not making much difference in observed performance. do we still want to cache it?

Reply:
I guess it's not too important if there's no noticeable perf improvement